lib/access-context.js

```diff
@@ -138,17 +146,17 @@ AccessContext.prototype.getUserId = function() {
 
     // the principalType must either be 'USER'
     if (p.type === Principal.USER) {
-      return p.id;
+      return {id: p.id, principalType: p.type};
     }
 
     // or permit to resolve a valid user model
     var userModel = this.registry.findModel(p.type);
     if (!userModel) continue;
     if (userModel.prototype instanceof BaseUser) {
-      return p.id;
+      return {id: p.id, principalType: p.type};
     }
   }
-  return null;
+  return {};
```

Comment:
turned the `getUserId` method in `getUser`, now returning an object of the like `{id: p.id, principalType: p.type}`
of course the `getUserId` is kept for backward compatibility just a few lines above, in a shortened form, reusing the `getUser` method